Formula/bacula-fd@9.4.rb

```diff
@@ -0,0 +1,68 @@
+class BaculaFdAT94 < Formula
+  desc "Network backup solution"
+  homepage "https://www.bacula.org/"
+  url "https://downloads.sourceforge.net/project/bacula/bacula/9.4.4/bacula-9.4.4.tar.gz"
+  sha256 "0fe37a02ca768a720099d0d03509c364aff2390c05544d663f4819f8e7fc20be"
+
+
+  keg_only :versioned_formula
+
+  depends_on "openssl@1.1"
+  depends_on "readline"
+
+  uses_from_macos "zlib"
+
+  def install
+    # * sets --disable-conio in order to force the use of readline
+    #   (conio support not tested)
+    # * working directory in /var/lib/bacula, reasonable place that
+    #   matches Debian's location.
+    system "./configure", "--prefix=#{prefix}",
+                          "--sbindir=#{bin}",
+                          "--with-working-dir=#{var}/lib/bacula",
+                          "--with-pid-dir=#{var}/run",
+                          "--with-logdir=#{var}/log/bacula",
+                          "--enable-client-only",
+                          "--disable-conio",
+                          "--with-readline=#{Formula["readline"].opt_prefix}"
+
+    system "make"
+    system "make", "install"
+
+    # Avoid references to the Homebrew shims directory
+    inreplace Dir[prefix/"etc/bacula_config"],
+              HOMEBREW_SHIMS_PATH/"mac/super/", ""
+
+    (var/"lib/bacula").mkpath
+  end
+
+  def post_install
+    (var/"run").mkpath
+  end
+
+  plist_options :startup => true, :manual => "bacula-fd"
+
+  def plist
+    <<~EOS
+      <?xml version="0.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{opt_bin}/bacula-fd</string>
+            <string>-f</string>
+          </array>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/bacula-fd -? 2>&1", 1)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This PR is an almost straight copy of [bacula-fd](https://github.com/Homebrew/homebrew-core/blob/master/Formula/bacula-fd.rb) already in the repository, and the code in question is identical to what is already in the repository.  I am not authorized to spend my time fixing the original committers omissions beyond what is required to get the software to work.

Reply:
In that case we better close this. It might be a straight copy, but every pull request is reviewed and sometimes someone spots things they didn't before.

Reply:
That's your prerogative, but does that mean you will be rejecting any further updates to the `bacula-fd` formula until the maintainer fixes the issue?

Reply:
We don't have special maintainers for software, the community maintains all of homebrew.

And if I notice a pull request that's more than just a version bump I'll ask the author to fix these things, yes.